src/pessimisticPatternMatcherUtils.cc

```diff
@@ -210,72 +196,65 @@ RotationTestResult test_rotation(ndarray::Array<double, 1, 1> const& src_center,
-    Eigen::Matrix<double, 3, 3> shift_matrix;
+    Eigen::Matrix3d shift_matrix;
     if (sin_shift > 0) {
         Eigen::Vector3d rot_axis = ndarray::asEigenMatrix(src_center).head<3>().cross(ref_center_eigen);
         rot_axis /= sin_shift;
         shift_matrix = create_spherical_rotation_matrix(rot_axis, cos_shift, sin_shift);
     } else {
-        shift_matrix = Eigen::Matrix<double, 3, 3>::Identity();
+        shift_matrix = Eigen::Matrix3d::Identity();
     }
 
-    // Now that we have our shift we apply it to the src delta vector
-    // and check the rotation.
-    Eigen::Vector3d rot_src_delta = shift_matrix * ndarray::asEigenMatrix(src_delta).head<3>();
+    // Now that we have our shift we apply it to the src delta vector and check the rotation.
+    Eigen::Vector3d rot_src_delta = shift_matrix * ndarray::asEigenMatrix(src_delta);
     Eigen::Vector3d proj_src_delta = rot_src_delta - rot_src_delta.dot(ref_center_eigen) * ref_center_eigen;
-    Eigen::Vector3d ref_delta_eigen = ndarray::asEigenMatrix(ref_delta).head<3>();
+    Eigen::Vector3d ref_delta_eigen = ndarray::asEigenMatrix(ref_delta);
 
     Eigen::Vector3d proj_ref_delta =
             ref_delta_eigen - ref_delta_eigen.dot(ref_center_eigen) * ref_center_eigen;
     double proj_src_delta_sq = proj_src_delta.dot(proj_ref_delta);
     double cos_rot_sq = proj_src_delta_sq * proj_src_delta_sq /
                         (proj_src_delta.dot(proj_src_delta) * proj_ref_delta.dot(proj_ref_delta));
-    // If the rotation isn't in tolerance return an empty vector.
-    RotationTestResult result;
     if (cos_rot_sq < max_cos_rot_sq) {
-        return result;
+        // failed test
+        return RotationTestResult();
     }
-    result.cos_rot_sq = cos_rot_sq;
-    result.proj_ref_ctr_delta = proj_ref_delta;
-    result.shift_matrix = shift_matrix;
-    result.success = true;
-
-    return result;
+    return RotationTestResult(cos_rot_sq, proj_ref_delta, shift_matrix);
 }
 
-Eigen::Matrix<double, 3, 3> create_spherical_rotation_matrix(Eigen::Vector3d const& rot_axis,
-                                                             double cos_rotation, double sin_rotation) {
-    Eigen::Matrix<double, 3, 3> rot_cross_matrix;
-    rot_cross_matrix << 0., -rot_axis[2], rot_axis[1], rot_axis[2], 0., -rot_axis[0], -rot_axis[1],
-            rot_axis[0], 0.;
-    Eigen::Matrix<double, 3, 3> shift_matrix = cos_rotation * Eigen::Matrix<double, 3, 3>::Identity() +
-                                               sin_rotation * rot_cross_matrix +
-                                               (1 - cos_rotation) * rot_axis * rot_axis.transpose();
+Eigen::Matrix3d create_spherical_rotation_matrix(Eigen::Vector3d const& rot_axis, double cos_rotation,
+                                                 double sin_rotation) {
+    // clang-format off
+    Eigen::Matrix3d rot_cross_matrix;
+    rot_cross_matrix << 0., -rot_axis[2], rot_axis[1],
```

Comment:
I'd move the "off" comment to just before `rot_cross_matrix <<`. There's no reason clang-format shouldn't touch the variable declaration...